src/providers/Dao.js

```diff
@@ -0,0 +1,51 @@
+import React, { useContext } from 'react'
+import { addressesEqual } from '@1hive/1hive-ui'
+import { useLocation } from 'react-router-dom'
+import daoList from '@1hive/gardens-dao-list'
+
+import { useWallet } from '../providers/Wallet'
+
+const DAOContext = React.createContext()
+
+export function parsePath(pathname, search = '') {
+  const [, ...parts] = pathname.split('/')
+
+  return parts[0]
+}
+
+function getConnectedAccountBalance(account, dao) {
+  return 0
+}
+
+export function DAOProvider({ children }) {
+  const location = useLocation()
+  const { account } = useWallet()
+
+  // get all the daos for the dashboard
+
+  // note that i am calling this daoId because at some point we might want to also handle aragon id's and not just addresses
+  const firstParameter = parsePath(location.pathname)
```

Comment:
Now that we are using the :daoId` match, we can yse `useRouteMatch`, adding what i suggested of having `garden` as prefix path we can do

```
  const match = useRouteMatch('/garden/:daoId')
```